packages/dashmate/test/e2e/tests/testnetMasternode.spec.js

```diff
@@ -0,0 +1,257 @@
+      }
+
+      if (configEnvs.CORE_MASTERNODE_ENABLE !== 'true'
+        || !configEnvs.CORE_MASTERNODE_OPERATOR_PRIVATE_KEY) {
+        throw new Error('This is not masternode configuration. Core masternode parameter is disabled.')
+      }
+    });
+
+    it('Start testnet nodes', async () => {
+      await dashmate.start(testnetNetwork);
+
+      await isTestnetServicesRunning(true, container);
+    });
+
+    it('Check core status before core sync process finish', async () => {
+      const coreStatus = await dashmate.checkStatus('core');
+      const coreOutput = JSON.parse(coreStatus.toString());
+
+      //update core version in config because versions don't match??
+      // const coreVersion = core.docker.image.replace(/\/|\(.*?\)|dashpay|dashd:|\-(.*)/g, '');
+      // expect(output.latestVersion).to.be.equal(coreVersion)
+      // try {
+      //   const latestVersionRes = await fetch('https://api.github.com/repos/dashpay/dash/releases/latest');
+      //   const latestVersion = (await latestVersionRes.json()).tag_name.substring(1);
+      //   expect(output['Latest version']).to.be.equal(latestVersion)
+      // } catch (e) {
+      //   throw e;
+      // }
+
+      expect(coreOutput.network).to.be.equal(STATUS.testnetNetwork);
+      expect(coreOutput.chain).to.be.equal(STATUS.testChain);
+      expect(coreOutput.dockerStatus).to.be.equal('running');
+      expect(coreOutput.syncAsset).to.be.equal(MasternodeSyncAssetEnum.MASTERNODE_SYNC_BLOCKCHAIN);
+
+      const peersData = await execute('docker exec dash_masternode_testnet-core-1 dash-cli getpeerinfo').then((peers) => {
+        return JSON.parse(peers.toString());
+      });
+      const peersNumber = Object.keys(peersData).length;
+      expect(coreOutput.peersCount).to.be.equal(peersNumber, 'Peers number are not matching!');
+
+      let peerHeader;
+      let peerBlock;
+      do {
+        for (const peer of peersData) {
+          peerBlock = peer['synced_blocks'];
+          peerHeader = peer['synced_headers'];
+        }
+      } while (peerHeader < 1 && peerBlock < 1);
+
+      expect(coreOutput.p2pService).to.not.be.empty();
+      expect(coreOutput.p2pPortState).to.be.equal(PortStateEnum.CLOSED);
+      expect(coreOutput.rpcService).to.not.be.empty();
+
+      let coreSyncOutput;
+      do {
+        coreSyncOutput = JSON.parse(coreStatus.toString());
+      } while (+coreSyncOutput.blockHeight < 1 && +coreOutput.headerHeight < 1);
+
+      const explorerBlockHeightRes = await fetch(`${INSIGHT_URLs[testnetNetwork]}/status`);
+      ({info: {blocks: explorerBlockHeight}} = await explorerBlockHeightRes.json());
+      expect(+coreSyncOutput.remoteBlockHeight).to.be.equal(explorerBlockHeight);
+
+      expect(+coreSyncOutput.difficulty).to.be.greaterThan(0);
+
+      expect(coreOutput.serviceStatus).to.be.equal(ServiceStatusEnum.syncing);
+      if (!(coreOutput.verificationProgress > 0 && coreOutput.verificationProgress <= 100)) {
```

Comment:
In JSON output, verificationProgress measured from 0 to 1, as I remember

Reply:
Fixed, wait for next commit